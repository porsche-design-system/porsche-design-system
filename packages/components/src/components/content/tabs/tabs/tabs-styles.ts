import {getCss, pxToRemWithUnit, getThemedColors, getFocusStyles} from '../../../../utils';
import type { ThemeExtendedElectric } from '../../../../types';

export const getComponentCss = (theme: ThemeExtendedElectric): string => {
  const { baseColor } = getThemedColors(theme);
  const {
    '&:focus': focusStyle,
    '&:focus:not(:focus-visible)': focusNotFocusVisibleStyle,
    '&::-moz-focus-inner': mozFocusInnerStyle,
    ...defaultStyle
  } = getFocusStyles({color: baseColor, offset: 1}) as any;

  return getCss({
    ':host': {
      display: 'block',
    },
    root: {
      marginBottom: pxToRemWithUnit(8),
    },
    '::slotted': {
      '&(p-tabs-item)': defaultStyle,
      '&(p-tabs-item:focus)': focusStyle,
      '&(p-tabs-item:focus:not(:focus-visible))': focusNotFocusVisibleStyle,
    },
  });
};
