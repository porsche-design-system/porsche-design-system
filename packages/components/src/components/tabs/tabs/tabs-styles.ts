import { getCss } from '../../../utils';
import { pxToRemWithUnit } from '../../../styles';

export const getComponentCss = (): string => {
  return getCss({
    '@global': {
      ':host': {
        display: 'block',
      },
    },
    root: {
      marginBottom: pxToRemWithUnit(8),
    },
  });
};
