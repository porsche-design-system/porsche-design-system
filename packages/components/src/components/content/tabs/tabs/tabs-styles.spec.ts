import { getComponentCss } from './tabs-styles';

describe('getComponentCss()', () => {
  it.each<Parameters<typeof getComponentCss>>([['light'], ['dark'], ['light-electric']])(
    'should return correct css for theme: %s ',
    (...args) => {
      expect(getComponentCss(...args)).toMatchSnapshot();
    }
  );
});
