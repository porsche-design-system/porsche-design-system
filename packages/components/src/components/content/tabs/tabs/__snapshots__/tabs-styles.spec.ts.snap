// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentCss() should return correct css 1`] = `
":host {
  display: block;
}
.root {
  margin-bottom: 0.5rem;
}"
`;
